File: servers/sinatra/lib/schema_api.rb
require 'json'

MyApp.add_route('get', '/', {
  "resourcePath" => "/schema",
  "summary" => "Get model schemata for data objects returned by this API.",
  "nickname" => "find", 
  "responseClass" => "object", 
  "endpoint" => "/", 
  "notes" => "",
  "parameters" => [
    {
      "name" => "model",
      "description" => "Optional model filter. If omitted, will return all models.",
      "dataType" => "string",
      "paramType" => "query",
      "allowMultiple" => false,
      "allowableValues" => "",
      },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

MyApp.add_route('get', '/websocketHelp', {
  "resourcePath" => "/schema",
  "summary" => "Returns help text & subject list for websocket usage.",
  "nickname" => "websocketHelp", 
  "responseClass" => "object", 
  "endpoint" => "/websocketHelp", 
  "notes" => "",
  "parameters" => [
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: clients/csharp/src/Swagger/Client/Api/SchemaApi.cs
  using System;
  using System.Collections.Generic;
  using Swagger.Client.Common;
  using Swagger.Client.Model;
  namespace Swagger.Client.Api {
    public class SchemaApi {
      string basePath;
      private readonly ApiInvoker apiInvoker = ApiInvoker.GetInstance();

      public SchemaApi(String basePath = "https://www.bitmex.com/api/v1")
      {
        this.basePath = basePath;
      }

      public ApiInvoker getInvoker() {
        return apiInvoker;
      }

      // Sets the endpoint base url for the services being accessed
      public void setBasePath(string basePath) {
        this.basePath = basePath;
      }

      // Gets the endpoint base url for the services being accessed
      public String getBasePath() {
        return basePath;
      }

      /// <summary>
      /// Get model schemata for data objects returned by this API. 
      /// </summary>
      /// <param name="model">Optional model filter. If omitted, will return all models.</param>
      /// <returns></returns>
      public object find (string model) {
        // create path and map variables
        var path = "/schema".Replace("{format}","json");

        // query params
        var queryParams = new Dictionary<String, String>();
        var headerParams = new Dictionary<String, String>();
        var formParams = new Dictionary<String, object>();

        if (model != null){
          string paramStr = (model is DateTime) ? ((DateTime)(object)model).ToString("u") : Convert.ToString(model);
          queryParams.Add("model", paramStr);
		}
        try {
          if (typeof(object) == typeof(byte[])) {
            var response = apiInvoker.invokeBinaryAPI(basePath, path, "GET", queryParams, null, headerParams, formParams);
            return ((object)response) as object;
          } else {
            var response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams, formParams);
            if(response != null){
               return (object) ApiInvoker.deserialize(response, typeof(object));
            }
            else {
              return null;
            }
          }
        } catch (ApiException ex) {
          if(ex.ErrorCode == 404) {
          	return null;
          }
          else {
            throw ex;
          }
        }
      }
      /// <summary>
      /// Returns help text &amp; subject list for websocket usage. 
      /// </summary>
      /// <returns></returns>
      public object websocketHelp () {
        // create path and map variables
        var path = "/schema/websocketHelp".Replace("{format}","json");

        // query params
        var queryParams = new Dictionary<String, String>();
        var headerParams = new Dictionary<String, String>();
        var formParams = new Dictionary<String, object>();

        try {
          if (typeof(object) == typeof(byte[])) {
            var response = apiInvoker.invokeBinaryAPI(basePath, path, "GET", queryParams, null, headerParams, formParams);
            return ((object)response) as object;
          } else {
            var response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams, formParams);
            if(response != null){
               return (object) ApiInvoker.deserialize(response, typeof(object));
            }
            else {
              return null;
            }
          }
        } catch (ApiException ex) {
          if(ex.ErrorCode == 404) {
          	return null;
          }
          else {
            throw ex;
          }
        }
      }
      }
    }
